### PR TITLE
Implement window manipulation CSI sequences

### PR DESCRIPTION
**Pull request by nikitar020**
## Description

This PR implements CSI sequences for manipulating the Aminal window. I took the specification there: http://www.xfree86.org/4.7.0/ctlseqs.html (search for `Window manipulation` on the page)

Several sequences were **not** implemented:

- `CSI 6 t` - Lower the window to the bottom of the stacking order - Because the GLFW library does not have such a feature
- `CSI 7 t` - Refresh the window - Because Aminal continuously refreshes itself. By the way, it looks like on Windows this sequence is handled by the system.
- `CSI 19 t` - Report screen size in characters
- `CSI 20 t` - Report the window icon label - Because the GLFW library does not have such a feature
- `CSI 21 t` - Report the window title - Because the GLFW library does not have such a feature

Aminal logs reasonanble error message on each non-implemented sequence.

Fixes # 237 (partially)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

It was tested by manually issuing the control sequences.

**Test Configuration**:
* OS: `Windows` and `Linux`

### CODE REVIEW

File: terminal/terminal.go [hunk 1]
```diff
@@ -33,6 +33,25 @@ const (
 	MouseExtSGR
 )
 
+type WindowManipulationInterface interface {
+	RestoreWindow(term *Terminal) error
+	IconifyWindow(term *Terminal) error
+	MoveWindow(term *Terminal, pixelX int, pixelY int) error
+	ResizeWindowByPixels(term *Terminal, pixelsHeight int, pixelsWidth int) error
+	BringWindowToFront(term *Terminal) error
+	//SendWindowToBack(term *Terminal) error
```
Comment (MaxRis) [inline terminal/terminal.go]: Should commented function names be removed?

**Comment by MaxRis** [comment]
@nikitar020 strange, but can't see button to restart failed job in Travis

File: terminal/terminal.go (hunk 1 above)
Comment (nikitar020) [reply terminal/terminal.go]: @MaxRis Will remove them

**Comment by nikitar020** [comment]
@MaxRis It started to fail randomly after I merged recent changes from the `develop` branch.

**Comment by nikitar020** [comment]
@MaxRis So I re-run the build three times and it seems always succeeding. I think this PR can be merged.

**Comment by MaxRis** [comment]
@nikitar020 thanks!

**Pull request merged**